src/Merchants.php

```diff
@@ -142,15 +142,36 @@ public static function update_or_create_merchant() {
 			'merchant_name'    => $merchant_name,
 		);
 
-		// The response only contains the merchant id.
-		$response = API\Base::update_or_create_merchant( $args );
+		$cache_key = PINTEREST_FOR_WOOCOMMERCE_PREFIX . '_request_' . md5( wp_json_encode( $args ) );
+		$cache     = get_transient( $cache_key );
+
+		if ( false !== $cache ) {
+			return $cache;
```

Comment:
If we have cache value then it means that we are delaying next fetch. We can't return something here because the rest of the code will expect a correct value. We need to throw an exception to signalise to the rest of the App what is going on. We should throw it with the original code and a message that this is a backoff attempt.